src/Framework/Analyzer/test/Components/DisallowNonLiteralSequenceNumbersTest.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using Microsoft.AspNetCore.Analyzer.Testing;
+
+namespace Microsoft.AspNetCore.Analyzers.RenderTreeBuilder;
+
+public class DisallowNonLiteralSequenceNumbersTest
+{
+    private TestDiagnosticAnalyzerRunner Runner { get; } = new(new RenderTreeBuilderAnalyzer());
+
+    [Fact]
+    public async Task RenderTreeBuilderInvocationWithNumericLiteralArgument_Works()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Components.Rendering;
+var renderTreeBuilder = new RenderTreeBuilder();
+renderTreeBuilder.OpenElement(0, ""div"");
+renderTreeBuilder.CloseElement();
+";
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source);
+
+        // Assert
+        Assert.Empty(diagnostics);
+    }
+
+    [Fact]
+    public async Task RenderTreeBuilderInvocationWithNonConstantArgument_ProducesDiagnostics()
+    {
+        // Arrange
+        var source = TestSource.Read(@"
+using Microsoft.AspNetCore.Components.Rendering;
+var renderTreeBuilder = new RenderTreeBuilder();
+var i = 0;
+renderTreeBuilder.OpenRegion(/*MM*/i);
```

Comment:
Does the `/*MM*/` play a functional role here? Is this to test that the analyzer is looking for the first argument, and not the first bit of syntax within the parens?

Reply:
If we're keeping this, would it be OK to change it to something more descriptive like `/* Comment that the analyzer will skip */`?

Reply:
Our test infrastructure uses this to figure out where the analyzer produces the diagnostics at. It specifically looks for this token: https://github.com/dotnet/aspnetcore/blob/main/src/Analyzers/Microsoft.AspNetCore.Analyzer.Testing/src/TestSource.cs#L12